cdc/owner/scheduler.go

```diff
@@ -0,0 +1,359 @@
+		if taskWorkload == nil {
+			continue
+		}
+		for _, workload := range taskWorkload {
+			workloads[captureID] += workload.Workload
+		}
+	}
+
+	for _, pendingJob := range pendingJobs {
+		if pendingJob.TargetCapture == "" {
+			target, exist := s.moveTableTarget[pendingJob.TableID]
+			if !exist {
+				continue
+			}
+			pendingJob.TargetCapture = target
+			delete(s.moveTableTarget, pendingJob.TableID)
+			continue
+		}
+		switch pendingJob.Tp {
+		case schedulerJobTypeAddTable:
+			workloads[pendingJob.TargetCapture] += 1
+		case schedulerJobTypeRemoveTable:
+			workloads[pendingJob.TargetCapture] -= 1
+		default:
+			log.Panic("Unreachable, please report a bug",
+				zap.String("changefeed", s.state.ID), zap.Any("job", pendingJob))
+		}
+	}
+
+	getMinWorkloadCapture := func() model.CaptureID {
+		minCapture := ""
+		minWorkLoad := uint64(math.MaxUint64)
+		for captureID, workload := range workloads {
+			if workload < minWorkLoad {
+				minCapture = captureID
+				minWorkLoad = workload
+			}
+		}
+
+		if minCapture == "" {
+			log.Panic("Unreachable, no capture is found")
+		}
+		return minCapture
+	}
+
+	for _, pendingJob := range pendingJobs {
+		if pendingJob.TargetCapture != "" {
+			continue
+		}
+		minCapture := getMinWorkloadCapture()
+		pendingJob.TargetCapture = minCapture
+		workloads[minCapture] += 1
+	}
+}
+
+func (s *scheduler) syncTablesWithSchemaManager() []*schedulerJob {
+	var pendingJob []*schedulerJob
+	allTableListeningNow := s.table2CaptureIndex()
+	globalCheckpointTs := s.state.Status.CheckpointTs
+	for _, tableID := range s.allTableShouldBeListened {
+		if _, exist := allTableListeningNow[tableID]; exist {
+			delete(allTableListeningNow, tableID)
+			continue
+		}
+		// table which should be listened but not, add adding-table job to pending job list
+		boundaryTs := globalCheckpointTs
+		pendingJob = append(pendingJob, &schedulerJob{
+			Tp:         schedulerJobTypeAddTable,
+			TableID:    tableID,
+			BoundaryTs: boundaryTs,
+		})
+	}
+	// The remaining tables is the tables which should be not listened
+	tableShouldNotListened := allTableListeningNow
+	for tableID, captureID := range tableShouldNotListened {
+		opts := s.state.TaskStatuses[captureID].Operation
+		if opts != nil && opts[tableID] != nil && opts[tableID].Delete {
+			// the table is removing, skip
+			continue
+		}
+		pendingJob = append(pendingJob, &schedulerJob{
+			Tp:            schedulerJobTypeRemoveTable,
+			TableID:       tableID,
+			BoundaryTs:    globalCheckpointTs,
+			TargetCapture: captureID,
+		})
+	}
+	return pendingJob
+}
+
+func (s *scheduler) handleJobs(jobs []*schedulerJob) {
+	for _, job := range jobs {
+		job := job
+		s.state.PatchTaskStatus(job.TargetCapture, func(status *model.TaskStatus) (*model.TaskStatus, bool, error) {
+			switch job.Tp {
+			case schedulerJobTypeAddTable:
+				if status == nil {
+					// if task status is not found, we can just skip to set adding-table operation, this table will be added in next tick
```

Comment:
```suggestion
					// if task status is not found, we can just skip adding the adding-table operation, since this table will be added in the next tick
```